runtime/gc_base/modron.h

```diff
@@ -171,5 +171,13 @@ extern "C" fj9object_t j9gc_objaccess_tokenFromPointer(J9VMThread *vmThread, mm_
 
 /** @} */
 
+#if defined (OMR_GC_FULL_POINTERS)
+/* Enough space to hold a heap address, compressed or not */
+typedef UDATA MM_RememberedSetCard;
+#else /* OMR_GC_FULL_POINTERS */
+/* Compresses a heap address by shifting right by CARD_SIZE_SHIFT */
+typedef U_32 MM_RememberedSetCard; 
+#endif /* OMR_GC_FULL_POINTERS */
+
```

Comment:
How it suppose to work for run-time selected CR? For all other cases Remembered Set Cards is an array of `MM_RememberedSetCard[]`. So I would assume for run-time selected CR it is U_32 used in each U_64. But code below (add and subtract) suggests cards located contiguously and my assumption is wrong. Did I miss something?
Looks like Remembered Set Cards memory is used like an array of U_32 for CR and array of U_64 and for run-time CR it is not an array of `MM_RememberedSetCard[]` any more. Technically it would work but it is confusing for me slightly. Any opinions on this?

Reply:
Yes, it's represented as U32[] or UDATA[].  The MM_RememberedSetCard type is left for documentation purposes (much like fj9object_t), and the type has to be large enough to hold both sizes in mixed mode.